src/storage-client/src/controller.rs

```diff
@@ -3168,44 +3170,132 @@ where
         Ok(())
     }
 
-    /// On each element of `collections`, install a read hold on all of the `storage_dependencies`.
-    fn install_dependency_read_holds<I: Iterator<Item = GlobalId>>(
+    /// On each element of `collections`, install a read hold on all of the
```

Comment:
Seeing this again, I might say "For each element of `collections` ..." to make it clearer. But 🤷

Reply:
Sure thing